server/controllers/locationController.ts

```diff
@@ -0,0 +1,95 @@
+import { Request, Response } from 'express';
+import { Like } from 'typeorm';
+
+import { Location } from 'server/models/Location';
+
+export default {
+  async index(req: Request, res: Response) {
+    const { country_code, city, region, postal_code, address } = req.query;
+
+    const locations = await Location.find({
+      where: [
+        { country_code: Like(`%${country_code}%`) },
```

Comment:
Additionally, we should probably not have a `where` clause for any field that isn't included in the search query.

Reply:
Ok, I'll make it dynamic